qiskit/circuit/quantumcircuit.py

```diff
@@ -1960,6 +1960,22 @@ def power(
         power_circuit.append(gate.power(power, annotated=annotated), list(range(gate.num_qubits)))
         return power_circuit
 
+    @deprecate_arg(
+        name="annotated",
+        since="2.3",
+        additional_msg=(
+            "The method QuantumCircuit.control() no longer accepts `annotated=None`. "
+            "The new default is `annotated=True`, which represents the controlled gate "
+            "in the new quantum circuit as an `AnnotatedOperation` "
+            "(unless a dedicated controlled-gate class already exists). You can explicitly set "
+            "`annotated=False` to preserve the previous behavior. However, using `annotated=True` "
+            "is recommended, as it defers construction of the controlled circuit to transpiler, "
+            "and furthermore enables additional controlled-gate optimizations (typically leading "
+            "to higher-quality circuits)."
+        ),
+        predicate=lambda my_arg: my_arg is None,
+        removal_timeline="in Qiskit 3.0",
+    )
```

Comment:
same comment as above:
we add here the deprecation warning, but should we also change the default value of
` annotated: bool | None = None`, in line 1984 as part of this PR?
or only in 3.0?